extensions/wikia/Search/classes/MediaWikiService.php

```diff
@@ -1216,4 +1224,26 @@ public function getTitleFromPageId( $pageId ) {
 
 		return static::$pageIdsToTitles[$pageId];
 	}
+
+	/**
+	 * @param $pageId
+	 * @param $page
+	 * @return \Article
+	 */
+	private function useRedirection( $pageId, $page ): \Article {
```

Comment:
We might want to force type for `$page`.
```
There were 2 errors:
1) Wikia\Search\Test\MediaWikiServiceTest::testGetPageFromPageCanonicalArticle
Error: Call to a member function getTitle() on null
/home/travis/build/Wikia/app/includes/Article.php:165
/home/travis/build/Wikia/app/extensions/wikia/Search/classes/MediaWikiService.php:1235
/home/travis/build/Wikia/app/extensions/wikia/Search/classes/MediaWikiService.php:1121
/home/travis/build/Wikia/app/extensions/wikia/Search/tests/MediaWikiServiceTest.php:1178
2) Wikia\Search\Test\MediaWikiServiceTest::testGetPageFromPageRedirectArticle
Error: Call to a member function getTitle() on null
/home/travis/build/Wikia/app/includes/Article.php:165
/home/travis/build/Wikia/app/extensions/wikia/Search/classes/MediaWikiService.php:1235
/home/travis/build/Wikia/app/extensions/wikia/Search/classes/MediaWikiService.php:1121
/home/travis/build/Wikia/app/extensions/wikia/Search/tests/MediaWikiServiceTest.php:1233
```
 `Article::newFromId()` will return `null` if there is no such article in the database, so calling `Article::getTitle()` will always fail with a PHP Fatal Error.